src/components/ShareMyLocationButton.js

```diff
@@ -28,20 +28,33 @@ const ShareMyLocationButton = ({ disabled = false }) => {
 
   const handleShareLiveLocation = () => {
     if (navigator.geolocation) {
-      if (trackingId === null) {
-        const newTrackingId = navigator.geolocation.watchPosition(
-          handleSuccess,
-          handleError,
-        );
-        setTrackingId(newTrackingId);
-      }
+      const trackingId = navigator.geolocation.watchPosition(
+        handleSuccess,
+        handleError,
+      );
+
+      setTrackingId(trackingId);
+      setUserLocationShared(true);
     }
   };
 
+  const handleStopSharing = () => {
+    if (userLocationShared && trackingId) {
+      setTrackingId(navigator.geolocation.clearWatch(trackingId));
+    }
+    setUserLocationShared(false);
+    setNewCenterMap(defaultCenterMap);
+  };
+  console.log('---', trackingId);
   return (
-    <button onClick={handleShareLiveLocation} disabled={disabled}>
-      Share My Location
-    </button>
+    <>
+      <button onClick={handleShareLiveLocation} disabled={userLocationShared}>
+        Share My Location
+      </button>
+      {userLocationShared ? (
```

Comment:
Only for separate responsibilities, because it is ShareMyLocation (like a 'start' sharing location), it could be interesting to move the 'stop sharing' to a new component. But, I think it's good to question our mentors :).